### PR TITLE
Feature: RTL support + Improvements to font loading

### PR DESCRIPTION
**Pull request by ZakBrown93**
# Checklist
* [x] reviewed
* [x] tested

# Changes
- RTL languages supported in the PDFs exported from the application
- Makes use of an implementation of the BIDI algorithm provided by Twitter
- Fonts are now loaded as part of the client application rather than on request of a PDF
____
# Tests
- [x] See https://github.com/ripjar/kyc-ui/pull/730
____
# Dependencies
**None**

### CODE REVIEW

File: src/layoutBuilder.js [hunk 1]
```diff
@@ -661,12 +664,69 @@ LayoutBuilder.prototype.buildNextLine = function (textNode) {
 		return newInline;
 	}
 
+	function fixedCharCodeAt(str, idx) {
```
Comment (ZakBrown93) [inline src/layoutBuilder.js]: Is there an appropriate location for these to be installed as utilities in `pdfmake`?

File: src/textTools.js [hunk 2]
```diff
@@ -143,7 +142,7 @@ function copyStyle(source, destination) {
 	return destination;
 }
 
-function normalizeTextArray(array, styleContextStack) {
+function normalizeTextArray(array, styleContextStack, isRtl) {
```
Comment (ZakBrown93) [inline src/textTools.js]: ⚡️ `isRtl`

File: src/layoutBuilder.js [hunk 3]
```diff
@@ -700,6 +760,66 @@ LayoutBuilder.prototype.buildNextLine = function (textNode) {
 		isForceContinue = inline.noNewLine && !isHardWrap;
 	}
 
+	if (textNode.rtl) {
```
Comment (ZakBrown93) [inline src/layoutBuilder.js:826]: Let's tidy this logic up and update the comments (I won't remember what this does in 5 month's time).

**Review by baking-code — COMMENTED** [review]
Oh dear lord that style guide.

I don't feel I can properly review this without a phd

**Pull request merged**